operators/src/source/ogr_source.rs

```diff
@@ -0,0 +1,2856 @@
+                            .ok_or(Error::TimeIntervalColumnNameMissing)?,
+                    );
+
+                    TimeInterval::new(time_start, time_start + duration).map_err(Into::into)
+                })
+            }
+        })
+    }
+
+    fn initialize_types_and_builder(
+        dataset_information: &OgrSourceDataset,
+    ) -> (
+        HashMap<String, FeatureDataType>,
+        FeatureCollectionBuilder<G>,
+    ) {
+        let mut data_types = HashMap::new();
+        let mut feature_collection_builder = FeatureCollection::<G>::builder();
+        // TODO: what to do if there is nothing specified?
+        if let Some(ref column_spec) = dataset_information.columns {
+            // TODO: error handling instead of unwrap
+            for attribute in &column_spec.numeric {
+                data_types.insert(attribute.clone(), FeatureDataType::NullableNumber);
+                feature_collection_builder
+                    .add_column(attribute.clone(), FeatureDataType::NullableNumber)
+                    .unwrap();
+            }
+            for attribute in &column_spec.decimal {
+                data_types.insert(attribute.clone(), FeatureDataType::NullableDecimal);
+                feature_collection_builder
+                    .add_column(attribute.clone(), FeatureDataType::NullableDecimal)
+                    .unwrap();
+            }
+            for attribute in &column_spec.textual {
+                data_types.insert(attribute.clone(), FeatureDataType::NullableText);
+                feature_collection_builder
+                    .add_column(attribute.clone(), FeatureDataType::NullableText)
+                    .unwrap();
+            }
+        }
+        (data_types, feature_collection_builder)
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn compute_batch(
+        feature_iterator: &mut Peekable<FeatureIterator<'_>>,
+        feature_collection_builder: FeatureCollectionBuilder<G>,
+        dataset_information: &OgrSourceDataset,
+        default_geometry: &Option<G>,
+        data_types: &HashMap<String, FeatureDataType>,
+        query_rectangle: &QueryRectangle,
+        time_extractor: &dyn Fn(&Feature) -> Result<TimeInterval>,
+        chunk_byte_size: usize,
+    ) -> Result<FeatureCollection<G>> {
+        let mut builder = feature_collection_builder.finish_header();
+
+        for feature in feature_iterator {
+            if let Err(error) = Self::add_feature_to_batch(
+                default_geometry,
+                data_types,
+                &query_rectangle,
+                time_extractor,
+                &mut builder,
+                &feature,
+            ) {
+                match dataset_information.on_error {
+                    OgrSourceErrorSpec::Skip | OgrSourceErrorSpec::Keep => continue,
+                    OgrSourceErrorSpec::Abort => return Err(error),
+                }
+            }
+
+            if builder.byte_size() >= chunk_byte_size {
```

Comment:
I vote against it. This would be super-complicated to catch an edge case that might never be a problem. The semantics would also need to be defined. How much bigger than `byte_size` is too big, etc.

Reply:
I think it may be a problem some day if we only have a soft size bound, that is actually always violated. Imagine thousands of simultaneous requests that all are 1 MB over the specified limit... I don't say solve this right now but it is certainly feasible to calculate the size of a new feature and only add it to the chunk if it still fits into the size limit. Maybe it's difficult with the current builder but it is possible. That's why I vote for a todo or a task in phabricator.

Reply:
Then rows that are slightly bigger than chunk_bytes lead to emitted collections that are always empty. My idea was to stop adding stuff if the collection is above that threshold. We could have an RFC there.

Reply:
we need some clear semantics here, and right now chunk_byte_size says nothing about the size of the chunk. it can be much smaller, or much larger. its neither lower nor upper limit. its some value that most likely is a little lower than the average of the produced chunks. I'd like to have stronger guarantees here. You may create a RFC

Reply:
https://github.com/geo-engine/rfcs/issues/9